presto-salesforce/src/main/java/io/prestosql/plugin/salesforce/SalesforceClient.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    @Override
+    public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        int columnSize = typeHandle.getColumnSize();
+
+        String jdbcTypeName = typeHandle.getJdbcTypeName()
+                .orElseThrow(() -> new PrestoException(JDBC_ERROR, "Type name is missing: " + typeHandle));
+
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+
+        if (jdbcTypeName.equals("multipicklist")) {
+            VarcharType type = createVarcharType(typeHandle.getColumnSize());
+
+            if (typeHandle.getColumnSize() == 0 || typeHandle.getColumnSize() > VarcharType.MAX_LENGTH) {
+                type = createUnboundedVarcharType();
+            }
+
+            return Optional.of(blockMapping(new ArrayType(type), multiPicklistReadFunction(type), multiPicklistWriteFunction(type)));
+        }
+
+        switch (typeHandle.getJdbcType()) {
+            case Types.VARCHAR:
+            case Types.NVARCHAR:
+            case Types.LONGVARCHAR:
+            case Types.LONGNVARCHAR:
+                if (columnSize == 0 || typeHandle.getColumnSize() > VarcharType.MAX_LENGTH) {
+                    return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
+                }
+                return Optional.of(varcharColumnMapping(createVarcharType(columnSize)));
+            case Types.NUMERIC:
+            case Types.DECIMAL:
+                return Optional.of(doubleColumnMapping());
+            case Types.OTHER:
+                return Optional.of(ColumnMapping.sliceMapping(VarcharType.VARCHAR, otherReadFunction(), varcharWriteFunction()));
+            default:
+                return super.toPrestoType(session, connection, typeHandle);
+        }
+    }
+
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        return new SoqlQueryBuilder(identifierQuote).buildSql(this, session, connection, table.getCatalogName(), table.getSchemaName(), table.getTableName(), columns, table.getConstraint(), split.getAdditionalPredicate(), tryApplyLimit(table.getLimit()));
+    }
+
+    private static BlockReadFunction multiPicklistReadFunction(Type type)
+    {
+        return (resultSet, columnIndex) -> {
+            BlockBuilder builder = createUnboundedVarcharType().createBlockBuilder(null, 10);
+
+            for (String value : resultSet.getString(columnIndex).split(";")) {
+                type.writeSlice(builder, utf8Slice(value));
+            }
+
+            return builder.build();
+        };
+    }
+
+    private static BlockWriteFunction multiPicklistWriteFunction(Type type)
+    {
+        return (statement, index, value) -> {
+            // Not implemented
```

Comment:
Any reasons for not implementing it ? It would be iterating over that `value` and building a string ?

Reply:
Pure laziness at this point; the JDBC driver does not support writes yet.